lib/jelix-modules/jacl2db/classes/jAcl2DbManager.class.php

```diff
@@ -47,6 +47,10 @@ public static function addRight($group, $role, $resource = '-')
             return false;
         }
 
+        if (in_array($role, self::getHiddenRights())) {
```

Comment:
you must not do this check in jAcl2DbManager, else we couldn't any more to add or remove rights with the API. We want to hide rights from the user interface, not from the API. So you should such checks only in classes (controllers or elsewhere) that are related to the user interface management.

Reply:
The problem is that every existing rights that is not given to `jAcl2DbManager::setRightsOnGroup` will be unset and hiddenRights won't be given because they are not present anymore in the UI but we don't want to unset them. I can move the check to `setRightsOnGroup` but there is no way to not do this check somewhere in `jAcl2DbManager`

Reply:
You should modify jAcl2DbAdminUIManager. When giving rights to it, it should consolidate the list with hidden rights, before giving them to jAcl2DbManager. jAcl2DbManager should not have to do anything about hidden rights. So  `getHiddenRights()` method should move to jAcl2DbAdminUIManager (as not static method).